### PR TITLE
[libra-node] use chain ID from genesis, not config

### PR DESCRIPTION
**Pull request by christinacelee**
## Motivation

Preparation for removing chain ID from config 
In node initialization, read and use chain ID from genesis, not config

## Test Plan

Existing tests

### CODE REVIEW

File: libra-node/src/main_node.rs [hunk 1]
```diff
@@ -38,6 +42,21 @@ pub struct LibraHandle {
...
+        .map(|blob| {
+            AccountState::try_from(&blob).and_then(|state| {
+                Ok(state
+                    .get_chain_id_resource()?
+                    .ok_or_else(|| format_err!("ChainIdResource does not exist"))?
+                    .chain_id())
+            })
+        })
+        .ok_or_else(|| format_err!("Failed to fetch ChainIdResource"))?
+}
```
Comment (davidiw) [inline libra-node/src/main_node.rs:58]: I think we can use expects here.

File: libra-node/src/main_node.rs [hunk 2]
```diff
@@ -105,6 +124,10 @@ pub fn setup_environment(node_config: &NodeConfig) -> LibraHandle {
         "ChunkExecutor setup in {} ms",
         instant.elapsed().as_millis()
     );
+    let chain_id = match fetch_chain_id(&db_rw) {
+        Ok(chain_id) => chain_id,
+        Err(e) => panic!("ChainId missing in node initialization: {:?}", e),
+    };
```
Comment (davidiw) [inline libra-node/src/main_node.rs:130]: using expect above, simplifies this.

**Comment by christinacelee** [comment]
> why not remove chainid from the config

I'll do this in a separate PR, it's too scary until config-builder is fully deprecated everywhere

**Pull request closed without merging**